Maintenance: CSS flex guide: updates to guide 5 (#34237)
mdn/content: files/en-us/web/css/css_flexible_box_layout/controlling_ratios_of_flex_items_along_the_main_axis/index.md

This example demonstrates this `min-content` flooring, with the `flex-basis` resolving to the size of the content. If you change the width on the flex container — increasing it to `700px` for example — and then reduce the flex item width, you can see that the first two items will wrap, however they will never become smaller than their `min-content` size. When the box gets small, the space is only removed from the third item when shrunken further.

{{EmbedGHLiveSample("css-examples/flexbox/ratios/flex-shrink-min-content.html", '100%', 500)}}

In practice, this shrinking behavior provides reasonable results. It prevents content from disappearing completely and from getting smaller than their minimum content; the above rules are sensible for content needing to shrink to fit its container.

### Giving items different `flex-shrink` factors

In the same way as `flex-grow`, you can give flex-items different `flex-shrink` factors. This can help change the default behavior if, for example, you want an item to shrink more or less rapidly than its siblings or to not shrink at all.

> In this example, the first item has a `flex-shrink` factor of 1, the second `0` (so it won't shrink at all), and the third `4`, for a total of 5 shrink factors. The third item therefore shrinks approximately four times more rapidly than the first, but neither will shrink below their `min-content` width. Play around with the different values — as with `flex-grow`, you can use decimals or larger numbers here too.

Reviewer: ```suggestion
In this example, the first item has a `flex-shrink` factor of `1`, the second item `0` (so it won't shrink at all), and the third item `4`, making a total of `5` shrink factors. The third item, therefore, shrinks approximately four times more rapidly than the first, but neither will shrink below their `min-content` width. Play around with the different values: as with `flex-grow`, you can use decimals or larger numbers here as well.
```

{{EmbedGHLiveSample("css-examples/flexbox/ratios/flex-shrink-ratios.html", '100%', 570)}}

## Mastering sizing of flex items

> To really understand how flex item sizing works, you need to understand all these factors that come into play. Consider the following aspects, which we have already discussed in these guides:

Reviewer: these two can be combined:
```suggestion
To understand how flex item sizing works, you need to consider the factors below, which we've discussed in these guides:
```

> ### What sets the base size of the item?

Reviewer: ```suggestion
### What determines the base size of an item?
```

> 1. Is `flex-basis` set to `auto`, and does the item have a width set? If so, the size will be based on that width.

Reviewer: this does not need to be a numbered list IMO
```suggestion
- Is `flex-basis` set to `auto`, and does the item have a width set? If so, the size will be based on that width.
```

> 2. Is `flex-basis` set to `auto`? If so, the size is based on the item size.
> 3. Is `flex-basis` a length or percentage, but not zero? If so this is the size of the item (floored at `min-content`).
> 4. Is `flex-basis` set to `0`? If so, the item's size is not taken into consideration for the space-sharing calculation.

Reviewer: ```suggestion
- Is `flex-basis` set to `auto`? If so, the size will be based on the item's content size.
- Is `flex-basis` a length or a percentage, but not zero? If so, this will be the size of the item (floored at `min-content`).
- Is `flex-basis` set to `0`? If so, the item's size will not be taken into consideration for the space-sharing calculation.
```

> ### Do we have available space?

Reviewer: ```suggestion
### Is there available space?
```

> Items can't grow with no positive free space, and they won't shrink unless there is negative free space.

Reviewer: ```suggestion
Items can't grow without positive free space, and they won't shrink unless there is negative free space.
```

> 1. If we add up the widths of all the items (or heights if working in a column), is that total **less** than the total width (or height) of the container? If so, then you have positive free space and `flex-grow` comes into play.
> 2. If we add up the widths of all the items (or heights if working in a column), is that total **more** than the total width (or height) of the container? If so, you have negative free space and `flex-shrink` comes into play.

Reviewer: ```suggestion
- If we add up the widths of all the items (or heights if working in a column), is that total **less** than the total width (or height) of the container? If so, there will be positive free space, and `flex-grow` will come into play.
- If we add up the widths of all the items (or heights if working in a column), is that total **more** than the total width (or height) of the container? If so, there will be negative free space, and `flex-shrink` will come into play.
```

> ### Other ways to distribute space

Reviewer: ```suggestion
### What are the other ways to distribute space?
```

> If you do not want space added to the items, remember that you can deal with free space between or around items using the alignment properties described in the guide to aligning items in a flex container. The {{CSSxRef("justify-content")}} property will enable the distribution of free space between or around items. You can also use auto margins on flex items to absorb space and create gaps between items.

Reviewer: ```suggestion
If you do not want space added to the items, remember that you can manage free space between or around items using the alignment properties described in the guide for aligning items in a flex container. The {{CSSxRef("justify-content")}} property will enable the distribution of free space between or around items. You can also use auto margins on flex items to absorb space and create gaps between items.
```

> With all these flex properties available to you, you will find that most tasks can be achieved, although it might take a little bit of experimentation at first.

Reviewer: ```suggestion
With all these flex properties available to you, you will find that most layout tasks are possible, although it might take a little bit of experimentation at first.
```
